### ISSUE TITLE
reduce_lr_on_plateau_scheduler doesn't work

### ISSUE THREAD

**Issue by yinruiqing** [openspeech-team/openspeech#88]
## Environment info
- Python version: 3.8
- PyTorch version (GPU?): 1.7.1
- pytorch-lightning version: 1.4

  
## Information
Thank you for this wonderful project. Please take a look of [optimizer](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/optimizers.py#L154) and [optimizer_connector](https://github.com/PyTorchLightning/pytorch-lightning/blob/a64cc373946a755ce6c3aef57c1be607dfe29a0c/pytorch_lightning/trainer/connectors/optimizer_connector.py#L87) in pytorch lightning. The scheduler["reduce_on_plateau"] is always false and the monitor (val_loss) cannot be passed to step method. You can run any script with reduce_lr_on_plateau_scheduler  or warm_up_reduce_lr_on_plateau_scheduler to reproduce this bug.

**Comment by sooftware** [comment]
Thanks for letting us know. We'll check it. @hasangchun
